Lib/fontTools/ttLib/woff2.py

```diff
@@ -0,0 +1,1023 @@
+		assert hasattr(locaTable, 'locations') and hasattr(self, 'indexFormat'), \
+			"'getLocaData' must be run after 'reconstruct' method"
+		locations = locaTable.locations
+		indexFormat = self.indexFormat
+		if indexFormat == 0:
+			if max(locations) >= 0x20000:
+				raise TTLibError("indexFormat is 0 but local offsets > 0x20000")
+			if not all(l % 2 == 0 for l in locations):
+				raise TTLibError("indexFormat is 0 but local offsets not multiples of 2")
+			offsetArray = array.array("H")
+			for i in range(len(locations)):
+				offsetArray.append(locations[i] // 2)
+		else:
+			offsetArray = array.array("I", locations)
+		if sys.byteorder != "big":
+			offsetArray.byteswap()
+		return offsetArray.tostring()
+
+	def setLocaData(self, locaData, indexFormat, numGlyphs):
+		""" Decompile 'loca' table data using the specified 'indexFormat' and
+		'numGlyphs' (must be run before 'transform' method).
+		"""
+		self.indexFormat = self.ttFont['head'].indexToLocFormat = indexFormat
+		self.numGlyphs = self.ttFont['maxp'].numGlyphs = numGlyphs
+		self.ttFont['loca'].decompile(locaData, self.ttFont)
+
+	def transform(self, glyfData):
+		""" Convert the SFNT 'glyf' table data to WOFF2 transformed 'glyf' data. """
+		glyphOrder = ["glyph%d" % i for i in range(self.numGlyphs)]
+		self.ttFont.setGlyphOrder(glyphOrder)
+		self.ttFont.lazy = False
+
+		self.decompile(glyfData, self.ttFont)
+
+		self.nContourStream = b""
+		self.nPointsStream = b""
+		self.flagStream = b""
+		self.glyphStream = b""
+		self.compositeStream = b""
+		self.bboxStream = b""
+		self.instructionStream = b""
+		bboxBitmapSize = ((self.numGlyphs + 31) >> 5) << 2
+		self.bboxBitmap = array.array('B', [0]*bboxBitmapSize)
+
+		for glyphID in range(self.numGlyphs):
+			self._encodeGlyph(glyphID)
+
+		self.bboxStream = self.bboxBitmap.tostring() + self.bboxStream
+
+		self.version = 0
+
+		self.nContourStreamSize = len(self.nContourStream)
+		self.nPointsStreamSize = len(self.nPointsStream)
+		self.flagStreamSize = len(self.flagStream)
+		self.glyphStreamSize = len(self.glyphStream)
+		self.compositeStreamSize = len(self.compositeStream)
+		self.bboxStreamSize = len(self.bboxStream)
+		self.instructionStreamSize = len(self.instructionStream)
+
+		transfomedGlyfData = sstruct.pack(woff2GlyfTableFormat, self) + \
+			self.nContourStream + self.nPointsStream + self.flagStream + \
+			self.glyphStream + self.compositeStream + self.bboxStream + \
+			self.instructionStream
+		return transfomedGlyfData
+
+	def _encodeGlyph(self, glyphID):
+		glyphName = self.getGlyphName(glyphID)
+		glyph = self.glyphs[glyphName]
+		self.nContourStream += struct.pack(">h", glyph.numberOfContours)
+		if glyph.numberOfContours == 0:
+			return
+		elif glyph.isComposite():
+			self._encodeComponents(glyph)
+		else:
+			self._encodeCoordinates(glyph)
+		self._encodeBBox(glyphID, glyph)
+
+	def _encodeComponents(self, glyph):
+		lastcomponent = len(glyph.components) - 1
+		more = 1
+		haveInstructions = 0
```

Comment:
Isn't it the same thing in Python?
I took it from here:
https://github.com/behdad/fonttools/blob/master/Lib/fontTools/ttLib/tables/_g_l_y_f.py#L421